packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import { injectGlobal } from '@leafygreen-ui/emotion';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmall: 'xsmall',
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const defaultSpacing = 18;
+
+const mainContainer = css`
+  animation: fade-in 250ms ease-in-out;
+  -webkit-animation: fade-in 250ms ease-in-out;
+  background-color: ${uiColors.black};
+  opacity: 0.6;
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`;
+
+const scrollContainer = css`
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  overflow-y: auto;
+  display: flex;
+  justify-content: center;
+`;
+
+const modalContentStyle = css`
+  margin: ${defaultSpacing}px auto;
+  padding: 30px;
+  color: ${uiColors.gray.dark3};
+  background-color: ${uiColors.white};
+  border: 1px solid ${uiColors.gray.light3};
+  border-radius: 3px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  pointer-events: all;
+  height: 200vh;
```

Comment:
We're setting a fixed height for the modal?

Reply:
No sorry that was for testing the scroll!!